docs/mcap/README.md

```diff
@@ -0,0 +1,371 @@
+# MCAP File Format Specification
+
+[TLV wiki]: https://en.wikipedia.org/wiki/Type-length-value
+[profiles]: ./profiles
+[compression formats]: ./compression/supported-compression-formats.md
+[Diagram Unchunked]: ./diagrams/unchunked.png
+[Diagram Chunked]: ./diagrams/chunked.png
+
+> Status: DRAFT
+
+## Overview
+MCAP is a container file format for append-only storage of
+heterogeneously-schematized data. It is inspired by the ROS1 bag format and is
+intended to support flexible serialization options, while also generalizing to
+non-ROS systems and retaining characteristics such as self-containment and
+chunk compression. Features include:
+* Single-pass, indexed writes (no backward seeking)
+* Flexible message serialization options (e.g. ros1, protobuf, …)
+* Self-contained (message schemas are included in the file)
+* Fast remote file summarization
+* File attachments
+* Optional chunk compression
+* Optional CRC integrity checks
+
+More details on the requirements behind this design, and alternatives
+considered, can be found in Evaluation of robotics data recording file formats
+[TODO link].
+
+### Glossary
+Some helpful terms to understand in the following sections are:
+* Record: A [TLV triplet][TLV wiki] with type and value corresponding to one
+  of the opcodes and schemas below.
+* Topic: A named message type and associated schema.
+* Channel: A logical stream that contains messages on a single topic.
+  Channels are associated with a numeric ID by the recorder - the *Channel ID*.
+* Channel Info: A type of record describing information about a channel,
+  notably containing the name and schema of the topic.
+* Message: A type of record representing a timestamped message on a channel
+  (and therefore associated with a topic/schema). A message can be parsed by a
+  reader that has also read the channel info for the channel on which the
+  message appears.
+* Chunk: A record type that wraps a compressed set of channel info and
+  message records.
+* Attachment: Extra data that may be included in the file, outside the
+  chunks. Attachments may be quickly listed and accessed via an index at the
+  end of the file.
+* Index: The format contains indexes for both messages and attachments. For
+  messages, there are two levels of indexing - a *Chunk Index* at the end of
+  the file points to chunks by offset, enabling fast location of chunks based
+  on topic and timerange. A second index - the *Message Index* - after each
+  chunk contains, for each channel in the chunk, and offset and timestamp for
+  every message to allow fast location of messages within the decompressed
+  chunk data.
+
+  The attachment index at the end of the file allows for fast listing and
+  location of attachments based on name, timestamp, or attachment type.
+
+## Format Description
+An MCAP file is physically structured as a series of concatenated, type- and
+length-prefixed _"records"_, capped on each end with magic bytes:
```

Comment:
I might go with bold instead of italic for various terms we define; it visually stands out more.

Reply:
definitely agree